components/tidb_query_datatype/src/codec/mysql/set.rs

```diff
@@ -68,6 +88,12 @@ pub struct SetRef<'a> {
     value: usize,
 }
 
+impl<'a> SetRef<'a> {
+    pub fn new(data: &'a BufferVec, value: usize) -> Self {
+        Self { data, value }
+    }
+}
```

Comment:
I suggest adding something to check is a given bit is set. e.g. `is_set(0)` checks if the first element is set.